infrastructure/sam/MoralPainFunctions/src/main/java/edu/uva/cs/moralpain/typedb/TypeDBPost.java

```diff
@@ -0,0 +1,86 @@
+package edu.uva.cs.moralpain.typedb;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+
+import com.vaticle.typedb.client.TypeDB;
+import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBSession;
+import com.vaticle.typedb.client.api.TypeDBTransaction;
+import com.vaticle.typedb.client.api.answer.ConceptMap;
+import com.vaticle.typeql.lang.TypeQL;
+import com.vaticle.typeql.lang.query.TypeQLInsert;
+import static com.vaticle.typeql.lang.TypeQL.var;
+
+public class TypeDBPost implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    JSONObject obj = new JSONObject();
+    List<JSONObject> reports = new ArrayList<JSONObject>();
+    JSONParser parser = new JSONParser();
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("X-Custom-Header", "application/json");
+
+        APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+
+        String body = input.getBody();
+
+        try {
+            JSONObject jsonBodyObject = (JSONObject) parser.parse(body);
+            String id = (String) jsonBodyObject.get("id");
+            @SuppressWarnings("unchecked")
+            List<String> selections = (List<String>) jsonBodyObject.get("selections");
+            Long score_long = (Long) jsonBodyObject.get("score");
+            int score = score_long.intValue();
+            // date comes in as UNIX time in seconds, so we can use int. If it came in
+            // miliseconds, use long.
+            Long timestamp = (Long) jsonBodyObject.get("timestamp");
+
+            String ip = String.format("%s:1729", System.getenv("EC2_IP_ADDRESS"));
```

Comment:
checkout the variable manager class we have. It's useful for testing.

Reply:
https://github.com/kevinsullivan/moralpain_dev/pull/134/commits/23a225cc84ac2e019d418f87076de5678315d1cd